### DISCUSSION TITLE
How to convert the type "content" to "boolean"?

### DISCUSSION THREAD

**Discussion by IuvenisSapiens** [typst/typst#3683] [General]
### Considering the following typst snippet, I want to know how to convert the type _"content"_ to _"boolean"_.
```
#let FA = counter("FA")
#FA.step()
#FA.step()
#FA.step()
#FA.step()
#FA.step()
#FA.step()
#FA.step()
#FA.step()
#let judge_code = locate(loc => FA.final(loc).at(0))>= 6)
#judge_code // true
#type(judge_code) // content
```

**Comment by sitandr** [comment]
Well, you can't do it, this is just how the context system works. 

The content you see is not any value, instead, it is _an instruction_ to display (locate) in that part of the document the _result_ of the function. If you want to work with the _value_ of the function (somehow work with this bool), you should do that _inside_ the instruction. That way, everything that would depend on the state value would be located inside of the `locate`.

There are certain reasons to do it that way, see the state docs. I also recommend trying new `context` system for a clearer code instead of `locate`s. Maybe that would help to understand the way it works. Everything that can depend on the changing context (state in our case) should be contained in `context`.

By the way, this is not a Typst issue, it is a question. Hence, please, ask them either in Githib Discussions or Discord corresponding channels.

**Reply by IuvenisSapiens** [reply]
Thank you for your reply, which helps me get the desired code snippet:
```
#let FA = counter("FA")
#FA.step()
#FA.step()
#FA.step()
#FA.step()
#FA.step()
#FA.step()
<alloc>
#FA.step()
#FA.step()
#context if (FA.at(<alloc>).at(0))>= 6) { The number is greater than or equal to six. } else { The number is less than six. }
```
